source/Calamari.Tests/AWS/CloudFormation/CloudFormationFixture.cs

```diff
@@ -0,0 +1,182 @@
+﻿using Calamari.Common.Features.Packages;
+using Calamari.Common.Features.Processes;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Variables;
+using Amazon.CloudFormation;
+using Amazon.CloudFormation.Model;
+using System.Threading.Tasks;
+#if AWS
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using Amazon;
+using Amazon.Runtime;
+using Amazon.S3;
+using Calamari.Aws.Commands;
+using Calamari.Aws.Deployment;
+using Calamari.Aws.Integration.CloudFormation;
+using Calamari.Aws.Integration.S3;
+using Calamari.Aws.Serialization;
+using Calamari.Integration.FileSystem;
+using Calamari.Serialization;
+using Calamari.Tests.Fixtures.Deployment.Packages;
+using Calamari.Tests.Helpers;
+using FluentAssertions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using NUnit.Framework;
+using Octopus.CoreUtilities.Extensions;
+using Octostache;
+
+namespace Calamari.Tests.AWS
+{
+    [TestFixture, Explicit]
```

Comment:
These were based off existing S3 tests that are marked as Explicit and aren't run in CI.

To run these in CI would involve changing to the AWS OctopusAPITester credentials instead of the specific Calamari ones here, plus would require changing the access of those credentials to allow for CloudFormation deployments. We already have e2e tests for CloudFormation over in server, I'm undecided on whether we should:

- Keep these as is as they are helpful for debug style testing
- Throw these away altogether
- Move the e2e CloudFormation tests over to here in some form

Reply:
I would say remove the E2E tests in Server and test exclusively in Calamari

Reply:
To get these running in CI I've changed over to using the OctopusAPITester user, but I'll need to add some additional permissions to allow it to perform CloudFormation and S3 actions, do you foresee any issues with adding these @johnsimons ? Do I need to alert anyone else that I'm doing this?

Reply:
I don't think that is a problem

Reply:
These are now running successfully in CI :)